### DISCUSSION TITLE
Small issue with client.wait_for

### DISCUSSION THREAD

**Discussion by Martyn0324** [Rapptz/discord.py#5963] [Q&A] [answered]
I'm having a small problem with `client.wait_for` feature for a minigame I'm developing for my bot. Upon initiating the minigame using a command, the bot asks the user to make a choice and then have to wait for user's response.
If the user answer the bot right on next message, things works fine. However, if the user answer with something other than the desired response, then the bot will just cancel everything. Is there a way to avoid this?

**Example - Working just fine**
_User:_ $choice
_Bot:_ Ok. Choose 1 or 2
_User:_ 2
_Bot:_ Is this really your choice?´
_User:_ Yes
_Bot:_ Fine

**Example - Unwanted situation**
_User:_ $choice
_Bot:_ Ok. Choose 1 or 2
_User:_ 2
_Bot:_ Is this really your choice?
_User:_ Maybe
_User:_ Yes
(Command is cancelled and user has to start everything over again)

Here's the code I'm using:

```
#Irrelevant line: Just for context

 def confirmation(confirmation):
            return confirmation.author == opponent

#This is the one that really matters:

await ctx.send('Cráááá! Is this really your choice?')
            try:
                confirm = await client.wait_for('message', timeout=20, check = confirmation)
            except asyncio.TimeoutError:
                return await ctx.send("Crááááá! Are you gonna choose it or not?")
            else:
                if confirm.content.lower() == 'no':
                    await ctx.send("Crááááá! Then call me when you make up your mind. Don't waste my time! Crááááa!")
                if confirm.content.lower() == 'yes':
                    await ctx.send("Cráááááá! Very well, then. Now, it's " + str(ctx.author.mention) + "'s turn to choose!")
```

**Comment by Rapptz** [comment, marked as answer]
Your check should constrain the content you care about as well. 

```py
def check(message):
    return message.author == opponent and message.content.lower() in ('no', 'yes')
```

**Reply by Martyn0324** [reply]
Wow, so simple... Thanks!
